Sources/FunctionalTableData/CollectionView/SupplementaryConfig.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  SupplementaryConfig.swift
+//  FunctionalTableData
+//
+//  Created by Jason Kemp on 2021-10-01.
+//  Copyright © 2021 Shopify. All rights reserved.
+
+import UIKit
+
+public struct Supplementary<View>: CollectionSupplementaryItemConfig where View: UICollectionReusableView {
+	public let kind: ReusableKind
+	
+	public init(kind: ReusableKind) {
+		self.kind = kind
+	}
+	
+	public func register(with collectionView: UICollectionView) {
+		collectionView.register(View.self, forSupplementaryViewOfKind: kind.rawValue, withReuseIdentifier: View.reuseIdentifier)
+	}
+	
+	public func dequeueView(collectionView: UICollectionView, indexPath: IndexPath) -> UICollectionReusableView {
+		collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: View.reuseIdentifier, for: indexPath)
+	}
+	
+	public func update(_ view: UICollectionReusableView, collectionView: UICollectionView, forIndex index: Int) {
+		//intentionally blank
+	}
+	
+	public func isEqual(_ other: CollectionSupplementaryItemConfig?) -> Bool {
+		guard let other = other as? Supplementary<View> else { return false }
+		return kind == other.kind
+	}
+}
+
+public struct SupplementaryConfig<View, State>: CollectionSupplementaryItemConfig, Hashable where View: UIView & ConfigurableView, State: Hashable, View.State == State {
+	public static func ==(lhs: SupplementaryConfig<View, State>, rhs: SupplementaryConfig<View, State>) -> Bool {
+		lhs.state == rhs.state && lhs.kind == rhs.kind
+	}
+	
+	public let state: State
+	private let supplementary: Supplementary<ReusableSupplementaryView<View>>
+	
+	public init(kind: ReusableKind, state: State) {
+		self.supplementary = Supplementary<ReusableSupplementaryView<View>>(kind: kind)
+		self.state = state
+	}
+	
+	public func hash(into hasher: inout Hasher) {
+		hasher.combine(kind)
+		hasher.combine(state)
+	}
```

Comment:
Do you have a case for this in mind?

Reply:
Not that I can think, but some other users might.

Reply:
OK, took a look at this. 

To make equality work out for `SupplementaryConfig` and `IndexableSupplementaryConfig`, I need State to be a minimum of `Equatable`. I need equality for those config types to properly diff sections. I technically don't _need_ 
`Hashable` but it's not much more to ask for, and it may future proof us from any collection view changes.